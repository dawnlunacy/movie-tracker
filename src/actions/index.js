export const getMovies = movies => ({
  type: 'GET_MOVIES',
  movies
});

export const handleError = errorMessage => ({
  type: 'HANDLE_ERROR',
  errorMessage
});

export const isLoading = boolean => ({
  type: 'IS_LOADING',
  boolean
});

export const saveUser = currentUser => ({
  type: 'SAVE_USER',
  currentUser
});

export const logoutUser = currentUser => ({
  type: 'LOGOUT_USER',
  currentUser
});

export const saveNewFavorite = favorited => ({
  type: 'SAVE_NEW_FAVORITE',
  favorited
})

export const retrieveFavorited = favorited => ({
  type: 'RETRIEVE_FAVORITED',
  favorited
});
