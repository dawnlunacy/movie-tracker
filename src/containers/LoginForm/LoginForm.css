.login-form {
    align-items: center;
    background-color: #161616;
    background-image: url('../../images/red-curtain-&-empty-theatre.jpg');
    background-repeat: no-repeat;
    background-size: 100% 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 100vh;
    margin: auto;
    padding: 30px;
    width: 100vw;
}

.log-in-background {
    background-color: rgba(0, 0, 0, 0.55);
    border-radius: 15px;
}

input, label {
    display: block;
    margin: 20px;
    text-align: center;
}

input {
    border-radius: 5px;
    height: 35px;
    width: 425px;
}

h2 {
    font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

h3 {
    color: #CC0000;
}

.form-button {
    border-radius: 10px;
    height: 25px;
    margin: 10px;
    width: 100px;
}

.login-to-sign-up {
    margin-top: 10%;
}

.login-btn {
    border-radius: 10px;
    height: 25px;
    margin: 10px;
    width: 100px;
}

.login-btn:hover {
    background-color: #d3af37;
    transform: scale(1.2);

}
