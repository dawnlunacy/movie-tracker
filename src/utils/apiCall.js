import { validate } from "@babel/types";

export const fetchData = async(url)  => {
  const response = await fetch(url)
  if(response.ok) {
    const data  = await response.json()
    return data;
  } else {
    throw Error(response.statusText)
  }
}

export const loginUser = async (userInfo, url) => {
  const options = {
    method: 'POST',
    body: JSON.stringify(userInfo),
    headers: {
      'Content-Type': 'application/json'
    }
  }

  const response = await fetch(url, options)
    if (!response.ok) {
      throw Error(response.statusText)
    } 
    const validatedUser = await response.json()
    console.log("VUSER", validatedUser)
    return validatedUser
    
}
