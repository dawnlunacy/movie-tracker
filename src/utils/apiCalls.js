
export const fetchData = async(url)  => {
  const response = await fetch(url)
  if(response.ok) {
    const data  = await response.json()
    return data;
  } else {
    throw Error(response.statusText)
  }
}

export const getUser = async (userInfo, url) => {
  const options = {
    method: 'POST',
    body: JSON.stringify(userInfo),
    headers: {
      'Content-Type': 'application/json'
    }
  }

  const response = await fetch(url, options)
    return response
}

export const postFavorite = async (movieInfo, id) => {
  const options = {
    method: 'POST',
    body: JSON.stringify(movieInfo),
    headers: {
      'Content-Type': 'application/json'
    }
  }

  const response = await fetch(`http://localhost:3001/api/v1/users/${id}/moviefavorites`, options)
  const data = await response.json();
  console.log('data', data)
    return data
}
