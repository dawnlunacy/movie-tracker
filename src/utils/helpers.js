export const filteredMovieData = movies => {
  return movies.map(movie => ({
    id: movie.id,
    title: movie.title,
    poster: movie.poster_path,
    rating: movie.vote_average,
    overview: movie.overview
  }))
}