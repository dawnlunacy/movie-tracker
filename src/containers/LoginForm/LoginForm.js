import React, { Component } from 'react';
import { getUser } from '../../utils/apiCalls';
import { connect } from 'react-redux';
import { saveUser } from '../../actions/index';
import { Redirect } from 'react-router-dom';
import './LoginForm.css';

class LoginForm extends Component {
    constructor() {
        super()
        this.state = {
            userInput: {
                email: '',
                password: '',
            },
            error: '',
            isLoggedIn: false
        }
      }
  

    handleChange = (e) => {
        let userInput = this.state.userInput;
      userInput = {...userInput, [e.target.name]: e.target.value}
        this.setState({userInput: userInput})
    }
    
    submitForm = async (e) => {
        e.preventDefault();
        const userVerification = await getUser(this.state.userInput, 'http://localhost:3001/api/v1/login');
        this.validateResponse(userVerification)
        this.resetInputs()
    }

    validateResponse = async (response) => {
        const { saveUser } = this.props;
        if (!response.ok && response.status === 404) {
            this.setState({error: "There was a problem with the server. Please try again"})
          }
        if (!response.ok) {
            const error = await response.json()
            this.setState({error: error.error})
          } else {
          const newUser = await response.json()
          console.log("NEWUSER", newUser)
          saveUser(newUser);
          this.setState({isLoggedIn: true})
        }
      }

    resetInputs = () => {
      this.setState({
          userInput: {
              email: '',
              password: ''
          }
      })
    }


    render() {
        if (this.state.isLoggedIn) {
            return <Redirect to = '/' />
        }
        return (
            <>
            <form className="login-form">
                <div className="sign-up-background">
                <h2> Login </h2>
                <input
                    className="email-input"
                    type="text"
                    placeholder="Enter Email"
                    name="email"
                    value={this.state.email}
                    onChange={this.handleChange}
                />
                <input
                    className="password-input"
                    type="text"
                    placeholder="Enter Password"
                    name="password"
                    value={this.state.password}
                    onChange={this.handleChange}
                />
                <button onClick={(e) => this.submitForm(e)}> LOGIN </button>

                <p> {this.state.error} </p>
                </div>
            </form>

          
        </>
        )
    }
}
    
export const mapDispatchToProps = dispatch => ({
    saveUser: currentUser => dispatch(saveUser(currentUser))
})

export default connect(null, mapDispatchToProps)(LoginForm)


