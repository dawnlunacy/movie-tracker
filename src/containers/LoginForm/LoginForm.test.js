import React from 'react';
import { LoginForm, mapDispatchToProps } from './LoginForm';
import { shallow } from 'enzyme';
import { getUser } from '../../utils/apiCalls';
import { saveUser } from '../../actions/index';
// jest.mock('../../actions/index');

console.log('getUser', getUser)
console.log('saveUser', saveUser)

describe('LoginFormContainer', () => {
  describe('AddTodoForm component', () => {
    let wrapper;
    // mockSaveUser = jest.fn();

    beforeEach(() => {
      wrapper = shallow(<LoginForm />)
    })

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })
    
    it('should set state of UserInput email when handleChange is called with an event', () => {
      const emailEvent = {
        target: {
          name: "email",
          value: "rudd.lacy@gmail.com"
        }
      }
     
      wrapper.instance().handleChange(emailEvent);

      expect(wrapper.state('userInput').email).toEqual(emailEvent.target.value);
      })

    it('should set state of UserInput password when handleChange is called with an event', () => {

      const passwordEvent = {
        target: {
         name: "password",
         value: "password"
        }
      }

    wrapper.instance().handleChange(passwordEvent);

    expect(wrapper.state('userInput').password).toEqual(passwordEvent.target.value);
    })

    it('should reset state of errors when handleChange is called', () => {
      const defaultState = {
        userInput: {
            email: 'mooses@forest.io',
            password: 'MooseIsBetterThanPants',
        },
        error: 'Username or password incorrect',
        formReady: false,
        isLoggedIn: false
      }
      const expected = {
        userInput: {
            email: 'mooses@forest.io',
            password: 'MoosesAreWAYBetterThanPants',
        },
        error: '',
        formReady: false,
        isLoggedIn: false
      }

      const passwordEvent = {
        target: {
         name: "password",
         value: "MoosesAreWAYBetterThanPants"
        }
      }

      wrapper.instance().setState(defaultState);

      wrapper.instance().handleChange(passwordEvent);

      expect(wrapper.state()).toEqual(expected);
    })

    it('should reset state of userInputs when resetInputs is called', () => {
      const defaultState = {
        userInput: {
            email: 'mooses@forest.io',
            password: 'MooseIsBetterThanPants',
        },
        error: '',
        formReady: false,
        isLoggedIn: false
      }
      const expected = {
        userInput: {
            email: '',
            password: '',
        },
        error: '',
        formReady: false,
        isLoggedIn: false
      }

      wrapper.instance().setState(defaultState);

      wrapper.instance().resetInputs();

      expect(wrapper.state()).toEqual(expected);
    })

    it('should call checkInputsForValues when submitForm is called', () => {
      const mockEvent = { preventDefault: jest.fn() };
      // console.log("Wrapper", wrapper.instance())
      getUser.mockImplementation(() => {
        return Promise.resolve([{id: 2, name:'Lacy', email:'rudd.lacy@gmail.com'}])
      })
      wrapper.instance().sumbitForm(mockEvent);

      expect(wrapper.instance().checkInputsForValues).toHaveBeenCalled();
    })
  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with an saveUser action when validateResponse is called with a successful response', () => {
      const mockDispatch = jest.fn();
      const mockResponse = {
        "id": 3,
        "name": "Lacy",
        "email": "rudd.lacy@gmail.com"
    }

      const actionToDispatch = saveUser(mockResponse);

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.saveUser(mockResponse);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);

    })  
  })
})
