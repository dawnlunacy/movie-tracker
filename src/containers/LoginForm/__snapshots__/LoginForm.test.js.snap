// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFormContainer AddTodoForm component should match snapshot 1`] = `
<Fragment>
  <form
    className="login-form"
  >
    <div
      className="log-in-background"
    >
      <Link
        className="App-img-form"
        to="/"
      >
        <img
          alt="Logo"
          className="App-img-form"
          src="MovieTracker_font_wave.png"
        />
      </Link>
      <h2>
         Login 
      </h2>
      <input
        className="email-input"
        name="email"
        onChange={[Function]}
        placeholder="Enter Email"
        type="text"
      />
      <input
        className="password-input"
        name="password"
        onChange={[Function]}
        placeholder="Enter Password"
        type="password"
      />
      <button
        className="form-btn"
        onClick={[Function]}
      >
         LOGIN 
      </button>
      <h3>
         
         
      </h3>
      <div
        className="login-to-sign-up"
      >
        <h4
          className="prompt-to-sign-up"
        >
           Don't have an account? 
        </h4>
        <Link
          to="/signup"
        >
           
          <button
            className="sign-up-btn"
          >
             SIGN UP
          </button>
           
        </Link>
      </div>
    </div>
  </form>
</Fragment>
`;
